### PR TITLE
PR to solve #59 (Read all data files, using regular expressions)

### LINKED ISSUE #59: Read all datafiles in a folder
When loading a line database, [SpectrumFactory.load_databank](https://radis.readthedocs.io/en/latest/source/radis.lbl.loader.html#radis.lbl.loader.DatabankLoader.load_databank) can currently read a file, or a list of file. We should allow it to read an entire folder, or (better) paths with pattern matching. 

Examples:

```
path=PATH_TO_FOLDER/cdsd_*
path=PATH_TO_FOLDER/*
path=PATH_TO_FOLDER/*.par
```

Modifications would have to be done in the [lbl/loader.py](https://github.com/radis/radis/blob/develop/radis/lbl/loader.py) file.   For instance line [1283](https://github.com/radis/radis/blob/develop/radis/lbl/loader.py#L1283)

We could use the [fnmatch](https://docs.python.org/2/library/fnmatch.html#module-fnmatch) library 

- [ ] Implement the pattern
- [ ] Ensures it works as an argument in load_databank
- [ ] Ensures it works for paths given in the ~/.radis config file
- [ ] Add tests
- [ ] Add examples in documentation

### PR DESCRIPTION
**Pull request by adityabharadwaj198**
### Description

----------------
Due to some issues when trying to rebase (here #97 ) , I had to create a new branch and push commits onto that and subsequently make a new PR.

---------------------
This pull request is to address ... #59  reading all files in a folder, using regular expression.
example: path = PATH_TO_FOLDER/*.par

will read all the files having .par extension

Changes made in this pr work for all the regular expressions, they are case insensitive.

Regular expressions like *co2*.par will match files containing CO2 and co2 in their names.

Fixes #59

### CODE REVIEW

**Comment by erwanp** [comment]
Very clean code and git history, well done!

Can I suggest one last thing? This new feature will interest many people (@CorentinGrimaldi @ulyssed), so we should document it. Good way to learn how to edit the website. 

Please do as you believe is best. I would suggest:

- [ ] in the [calc_spectrum](https://radis.readthedocs.io/en/latest/source/radis.lbl.calc.html#radis.lbl.calc.calc_spectrum) docstring, `databank` says 
  > the path to a valid database file,
  
  we could add `(accepts wildcoards ``*`` to select multiple files)`. And also add a line in the Example that is below. 

  ```
  databank='PATH/TO/05_HITEMP2019.par'    # path to a file
  databank='PATH/TO/*CO2*.par'    # to get all files that have CO2 in their name   ```

- [ ] same in the [SpectrumFactory.load_databank()](https://radis.readthedocs.io/en/latest/source/radis.lbl.loader.html#radis.lbl.loader.DatabankLoader.load_databank) and [init_databank()](https://radis.readthedocs.io/en/latest/source/radis.lbl.loader.html#radis.lbl.loader.DatabankLoader.init_databank) methods ?

- [ ] And finally, on the website itself. The [configuration-file](https://radis.readthedocs.io/en/latest/lbl/lbl.html#configuration-file) uses the `path=` argument. We should also add an example there, for instance
```
[MY-HITEMP-CO2]                  #  your databank name: use this in calc_spectrum()
                                 #  or SpectrumFactory.load_databank()
path =  D:\Databases\HITEMP-CO2\hitemp_*    # To load all hitemp files directly
format = hitran                  #  'hitran' (HITRAN/HITEMP), 'cdsd-hitemp', 'cdsd-4000'
                                 # databank text file format. More info in
                                 # SpectrumFactory.load_databank function.
parfuncfmt = hapi                # calculate partition functions
```
To modify this page you need to edit the [docs/lbl/lbl.rst](https://github.com/radis/radis/blob/develop/docs/lbl/lbl.rst) file. It's RST format, exactly like the docstrings above.

**Comment by adityabharadwaj198** [comment]
>  so we should document it

Surely, I was waiting for your approval on the commit history thing. I modified the docstrings and the rst file. Please review :)

**Comment by erwanp** [comment]
(i don't get the codestyle format error locally >> force-merging  :)  )

File: radis/lbl/loader.py [hunk 1]
```diff
@@ -1284,7 +1288,8 @@ def _check_database_params(
 
         # Check input types are correct
         if isinstance(path, string_types):  # make it a list
-            path = [path]
+            path = get_files_from_regex(path)
```
Comment (erwanp) [inline radis/lbl/loader.py:1291]: Actually the get_files_from_regex is not called if there was a list initially @CorentinGrimaldi

**Pull request merged**